compiler/rustc_passes/src/intrinsicck.rs

```diff
@@ -413,7 +411,7 @@ impl<'tcx> ExprVisitor<'tcx> {
                             let msg = format!(
                                 "register class `{}` requires at least one of the following target features: {}",
                                 reg_class.name(),
-                                features.join(", ")
+                                features.iter().map(|f| f.as_str()).collect::<Vec<_>>().join(", ")
```

Comment:
Why is this necessary? The former code is definitely more readable. A good way to avoid an intermediate allocation here would be to write:

```
features.iter().map(...).intersperse(", ").collect::<String>()
```

Reply:
TIL